### PR TITLE
Add pytest and CI workflow from torchtune

### PR DESCRIPTION
**Pull request by wconstab**

### CODE REVIEW

File: .github/workflows/unit_test.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+defaults:
+  run:
+    shell: bash -l -eo pipefail {0}
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.8', '3.9', '3.10', '3.11']
```
Comment (wconstab) [inline .github/workflows/unit_test.yaml]: hmm do we care about this many versions?

File: .github/workflows/unit_test.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        with:
+          auto-update-conda: true
+          miniconda-version: "latest"
+          activate-environment: test
+          python-version: ${{ matrix.python-version }}
+      - name: Update pip
+        run: python -m pip install --upgrade pip
+      - name: Install dependencies
+        run: |
+          conda install pytorch
```
Comment (wconstab) [inline .github/workflows/unit_test.yaml]: what's the right way to use recent pytorch?

- pin a version?
- use 'latest nightly' (what's the pip command for that)

File: .github/workflows/unit_test.yaml (hunk 1 above)
Comment (lessw2020) [reply .github/workflows/unit_test.yaml]: I would vote no - 3.9 or 3.10 by itself should be plenty.  As data to back that, most partners are on 3.9 or 3.10 and I haven't really seen issues that only appear specific to one version of Python. 
Also, imo - CI should be fast and smooth...waiting for it to loop through all tests  and * 4 versions of Python before you know your checkin is good is counterproductive.

Comment (wanchaol) [reply .github/workflows/unit_test.yaml]: Yeah we can even only keep 3.10 IMO, to keep CI fast

File: .github/workflows/unit_test.yaml (hunk 2 above)
Comment (wanchaol) [reply .github/workflows/unit_test.yaml]: I think we should always install latest nightly to ensure no breakage from main pytorch update. we can figure the command here https://pytorch.org/get-started/locally/

Comment (lessw2020) [reply .github/workflows/unit_test.yaml]: I agree with Wanchao, latest nightly should be the ticket to ensure no breakage from other checkins.
Here's the code for pip with nightly (and cuda 12.1):
~~~
pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
~~~

**Pull request merged**